index.js

```diff
@@ -49,3 +50,8 @@ Object.assign(module.exports, {
   restore: recorder.restore,
   back,
 })
+
+if (isOn()) {
+  // Setting the Back mode "activates" Nock by overriding the global entries in the `http/s` modules.
+  back.setMode(process.env.NOCK_BACK_MODE || 'dryrun')
```

Comment:
I think this is doing the right thing, though this implementation seems like a smell. I suppose it's because Nock Back is a sort of separable and optional component which has nothing to do getting Nock set up, and here is used for a side effect.

I really appreciate your making this change, though wonder, is it possible to refactor this a bit further so the code here looks more like `if isOn() { activate() }`? Perhaps Nock Back could be lazily set up, either when it is first used or if `setMode()` is invoked directly.

Reply:
That would mean Back wouldn't be in "dryrun" mode when Nock is imported. What all would that break?
Also if `NOCK_BACK_MODE` is set, we'd still need the Back call somewhere.

Reply:
> That would mean Back wouldn't be in "dryrun" mode when Nock is imported. What all would that break?

Hmm, that's a good question. Back's behavior is undefined outside of calls to `nock.back()` though devs are probably used to today's behavior, which is that `NOCK_BACK_MODE` is honored. I guess that would have to be dealt with in a breaking change.

Okay, I guess it's fine for now to do this, though leet's add to the comment an explanation that (1) we always activate nock on import, overriding the globals, (2) if nock back is configured, we need to honor that setting for backward compatibility reasons, and (3) otherwise we rely on nock back's default initializing side effect.

Reply:
Omg leet's